### PR TITLE
Support for HTMLDjango with Non-Nested Language Option

### PR DESCRIPTION
**Pull request by Brick85**
This pull request introduces support for comments in the HTMLDjango template language. While the detection of nested languages and formatting comments accordingly is generally advantageous, there are cases where it may be counterproductive, especially in file types like HTMLDjango templates.

In HTMLDjango templates, it is preferable to utilize Django template comments not only within HTML but also in JavaScript and CSS. These comments effectively exclude code from rendering and prove beneficial. The proposed change becomes essential for commenting on template control statements, addressing a limitation in the existing functionality.

Please review and consider merging this pull request to enhance the plugin's compatibility with HTMLDjango templates.

### CODE REVIEW

**Comment by andresmrm** [comment]
Hi! @JoosepAlviste, any reason for not accepting this MR? I've just tested it and it seems to work. It's very important for Django templates.

**Comment by JoosepAlviste** [comment]
Hey! To be honest, I have kind of forgotten about this PR 😅 Thanks for the ping!

I can definitely see that we should handle the Django templates correctly. At first glance, it seemed like having a `not_nested_languages` solution was a bit too specific (only required for 1 filetype), so I was hoping that I could come up with a better approach.

If I understand it correctly, then we want to absolutely ignore any `html` nodes inside `htmldjango` files? So, we would always use the `htmldjango` commenting style (`{# %s #}`) in the whole file. It would be nice, though, if `<script>` tags still had the correct `commentstring`.

**Comment by andresmrm** [comment]
> To be honest, I have kind of forgotten about this PR 😅 Thanks for the ping!

It happens. :stuck_out_tongue_closed_eyes:

> It would be nice, though, if <script> tags still had the correct commentstring.

I'm not sure about that... The thing I like about `{# #}` comments is that those lines don't get to the final HTML sent to the client. Comments can have sensitive or embarrassing stuff. 😅
On the other side, using `//` in a `<script>` gets a proper highlight here at my current nvim setup, while `{# #}` no. Maybe that can be fixed configuring plugins...
Anyway, I think I still prefer the first option to the second.

**Comment by Brick85** [comment]
I agree that `not_nested_languages` seems a bit large. However, I'm not sure of a better alternative yet. The only other solution I can think of right now would be a hardcoded value 😅

I prefer to comment everything with `{# #}` in Django templates, even within `<script>` tags. I've been using my fork with this change and it works well in both Django templates and other file types.

While this solution works for me, I understand that having a configuration option specific to one file type might not be the best approach. Perhaps other template dialects could benefit from a similar option in the future.

File: lua/ts_context_commentstring/internal.lua [hunk 1]
```diff
@@ -54,7 +54,11 @@ function M.calculate_commentstring(args)
   local location = args.location or nil
 
   local node, language_tree =
-    utils.get_node_at_cursor_start_of_line(vim.tbl_keys(config.get_languages_config()), location)
+    utils.get_node_at_cursor_start_of_line(
+      vim.tbl_keys(config.get_languages_config()),
+      config.config.not_nested_languages,
+      slocation
```
Comment (JoosepAlviste) [inline lua/ts_context_commentstring/internal.lua:60]: A typo here:

```suggestion
      location
```

File: lua/ts_context_commentstring/config.lua [hunk 2]
```diff
@@ -114,6 +115,11 @@ M.config = {
     },
   },
 
+  not_nested_languages = {
+    -- Languages at wich nesting stops
```
Comment (JoosepAlviste) [inline lua/ts_context_commentstring/config.lua:119]: Typo:

```suggestion
    -- Languages at which nesting stops
```

File: lua/ts_context_commentstring/internal.lua [hunk 3]
```diff
@@ -54,7 +54,11 @@ function M.calculate_commentstring(args)
   local location = args.location or nil
 
   local node, language_tree =
-    utils.get_node_at_cursor_start_of_line(vim.tbl_keys(config.get_languages_config()), location)
+    utils.get_node_at_cursor_start_of_line(
+      vim.tbl_keys(config.get_languages_config()),
+      config.config.not_nested_languages,
```
Comment (JoosepAlviste) [inline lua/ts_context_commentstring/internal.lua:59]: Let's also add `not_nested_languages` to the config type, here-ish: https://github.com/JoosepAlviste/nvim-ts-context-commentstring/blob/main/lua/ts_context_commentstring/config.lua#L44

File: lua/ts_context_commentstring/utils.lua [hunk 4]
```diff
@@ -76,13 +76,18 @@ end
 ---
 ---@param only_languages string[] List of languages to filter for, all
 ---  other languages will be ignored.
+---@param not_nested_languages string[] List of languages which stop nesting
```
Comment (JoosepAlviste) [inline lua/ts_context_commentstring/utils.lua:79]: Looks like the type here should be `table<string, boolean>`, right?

**Review by JoosepAlviste — CHANGES_REQUESTED** [review]

**Comment by JoosepAlviste** [comment]
Let's merge this! I can't think of a better solution and this is pretty OK anyways.

Added a couple comments and the formatting check is also not passing. Once those things are fixed, then we can merge 👍

**Review by JoosepAlviste — APPROVED** [review]

**Comment by JoosepAlviste** [comment]
Thanks for the contribution! 🎉

**Comment by Brick85** [comment]
Thank you for merging! I really appreciate your work on this project!

**Pull request merged**